src/NodeAnalyzer/CallLikeExpectsThisBoundClosureArgsAnalyzer.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\NodeAnalyzer;
+
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\Closure;
+use PHPStan\Reflection\ExtendedParameterReflection;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\NodeTypeResolver\PHPStan\ParametersAcceptorSelectorVariantsWrapper;
+use Rector\Reflection\ReflectionResolver;
+
+final readonly class CallLikeExpectsThisBoundClosureArgsAnalyzer
+{
+    public function __construct(
+        private ReflectionResolver $reflectionResolver
+    ) {
+    }
+
+    /**
+     * @return Arg[]
+     */
+    public function getArgsUsingThisBoundClosure(CallLike $callLike): array
+    {
+        $args = [];
+        $reflection = $this->reflectionResolver->resolveFunctionLikeReflectionFromCall($callLike);
+
+        if ($callLike->isFirstClassCallable() || $callLike->getArgs() === []) {
+            return [];
+        }
+
+        if ($reflection === null) {
+            return [];
+        }
+
+        $scope = $callLike->getAttribute(AttributeKey::SCOPE);
+
+        if ($scope === null) {
+            return [];
+        }
+
+        $parametersAcceptor = ParametersAcceptorSelectorVariantsWrapper::select($reflection, $callLike, $scope);
+        $parameters = $parametersAcceptor->getParameters();
```

Comment:
I think verify if there is `Closure` as Arg first is better to avoid possible reflection check:

```php
$args = $callLike->getArgs();
$hasClosureArg = (bool) array_filter($args, fn (Arg $arg): bool => $arg->value instanceof Closure);

if (! $hasClosureArg) {
    return null;
}
```